### ISSUE TITLE
Incorrect warning/error message for platform 'Alpine' and version 'any' in Galaxy meta/main.yml

### ISSUE THREAD

**Issue by coglinev3** [ansible/ansible-lint#3067]
##### Summary

Ansible-lint version 6.13.1 generates an error message if in the file ```meta/main.yml``` Alpine Linux is specified with version ```any``` as supported platform.

Many of my Ansible roles support Alpine Linux. The older versions of ansible-lint never had any problems with this.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

ansible-lint 6.13.1, installed via pip

```
ansible-lint --version
ansible-lint 6.13.1 using ansible 2.14.2
```

##### STEPS TO REPRODUCE

Create a ```meta/main.yml``` file like following in a role:

```
galaxy_info:
  author: author-name
  role_name: role-name
  namespace: author-name
  description: test
  min_ansible_version: "2.14"
  platforms:
    - name: Alpine
      versions:
        - any
    - name: Ubuntu
      versions:
        - bionic
        - focal
        - jammy    
  galaxy_tags: 
    - test
```

Create a file ```.ansible-lint``` like following in a role:

```
skip_list:
  - experimental
```

##### Desired Behavior

No warning or error message for platform ```Alpine``` and version ```any``` from ansible-lint, because ```any``` is the only supported release for platform Alpine (see Ansible Galaxy API
https://galaxy.ansible.com/api/v1/platforms/?search=Alpine).

```
GET /api/v1/platforms/?search=Alpine

HTTP 200 OK
  Allow: GET, HEAD, OPTIONS
  Content-Type: application/json
  Vary: Accept

  {
    "count": 1,
    "next": null,
    "next_link": null,
    "previous": null,
    "previous_link": null,
    "results": [
        {
            "id": 100,
            "url": "[/api/v1/platforms/100/](https://galaxy.ansible.com/api/v1/platforms/100/)",
            "related": {},
            "summary_fields": {},
            "created": "2016-11-17T09:12:38.313536Z",
            "modified": "2016-11-17T09:12:38.313573Z",
            "name": "Alpine",
            "release": "any",
            "active": true
        }
    ]
}
```

With ansible-lint version 6.12.2 it works without problems.

##### Actual Behavior

```
schema[meta]: {'name': 'Alpine', 'versions': ['any']} is not valid under any of the given schemas
meta/main.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


Read documentation for instructions on how to ignore specific rule violations.

             Rule Violation Summary              
 count tag          profile rule associated tags 
     1 schema[meta] basic   core                 

Failed after min profile: 1 failure(s), 0 warning(s) on xx files.
```

##### Additional comment:

* Maybe the rule is in ansible-lint version 6.13.1 no longer marked as ```experimental```.

* Of course it would be better if for Alpine Linux the exact versions can be specified, for example 3.17 or 3.16, like for the other distributions. Unfortunately, this is currently not supported by the Ansible Galaxy API.

**Comment by konstruktoid** [comment]
See https://github.com/ansible/ansible-lint/issues/3049

**Comment by ssbarnea** [comment]
I wonder where does the "any" come from... I was not able to spot this value documented anywhere, only the "all".

**Comment by konstruktoid** [comment]
It comes from the API response, https://galaxy.ansible.com/api/v1/platforms/?search=ArchLinux

**Comment by coglinev3** [comment]
The problem should not be marked as fixed. It still occurs with ansible-lint version 6.15.0. 
```
schema[meta]: {'name': 'Alpine', 'versions': ['any']} is not valid under any of the given schemas
```
But for ArchLinux and Alpine Linux, the only valid version is still "any".

See galaxy api: 
https://galaxy.ansible.com/api/v1/platforms/?search=Alpine
or 
https://galaxy.ansible.com/api/v1/platforms/?search=ArchLinux
